python/lsst/ts/m2com/controller_cell.py

```diff
@@ -0,0 +1,475 @@
+        connection_pooling_time = 0.1
+        while not self.are_clients_connected() and (
+            (time.monotonic() - time_start) < self.timeout_connection
+        ):
+            await asyncio.sleep(connection_pooling_time)
+
+        if not self.are_clients_connected():
+            raise RuntimeError(
+                "Timeout in connection - Conection timeouted, connection "
+                f"failed or cannot connect. Host: {self.host}, ports: "
+                f"{port_command} and {port_telemetry}."
+            )
+
+    def run_task_event_loop(self, process_event, *args, **kwargs):
+        """Run the task of event loop.
+
+        Parameters
+        ----------
+        process_event : `func` or `coroutine`
+            Function to process the event. It must has a keyward argument of
+            "message" to receive the message as a dictionary.
+        *args : `args`
+            Arguments needed in "process_event" function call.
+        **kwargs : `dict`, optional
+            Additional keyword arguments in "process_event" function
+            call.
+        """
+        self._task_event_loop = asyncio.create_task(
+            self._event_loop(process_event, *args, **kwargs)
+        )
+
+    async def _event_loop(self, process_event, *args, **kwargs):
+        """Update and output event information from component.
+
+        Parameters
+        ----------
+        process_event : `func` or `coroutine`
+            Function to process the event. It must has a keyward argument of
+            "message" to receive the message as a dictionary.
+        *args : `args`
+            Arguments needed in "process_event" function call.
+        **kwargs : `dict`, optional
+            Additional keyword arguments in "process_event" function
+            call.
+        """
+
+        self.log.debug("Begin to run the event loop from component.")
+
+        is_coroutine_function = is_coroutine(process_event)
+
+        while self.run_loops:
+
+            try:
+                message = (
+                    self.queue_event.get_nowait()
+                    if not self.queue_event.empty()
+                    else await self.queue_event.get()
+                )
+
+            # When there is no instance in the self.queue_event, the get() will
+            # be used to wait for the new coming event. It is a blocking call
+            # and will get the asyncio.CancelledError if we cancel the
+            # self._task_event_loop.
+            except asyncio.CancelledError:
+                message = ""
+
+            # Process the event
+            if is_coroutine_function:
+                await process_event(*args, message=message, **kwargs)
+            else:
+                process_event(*args, message=message, **kwargs)
+
+            await asyncio.sleep(self.timeout)
+
+        self.log.debug("Component's event loop exited.")
+
+    def run_task_telemetry_loop(self, process_telemetry, *args, period=2, **kwargs):
```

Comment:
I think this method name is a bit misleading. It is not actually "running" the task is it? but rather "starting" it in the background. Consider renaming it to reflect that. maybe `start_telemetry_loop_task`?

Reply:
Renamed in the commit: [449e880](https://github.com/lsst-ts/ts_m2com/pull/10/commits/449e8806e72c01fd253fbbdfc896198514aad657). Thanks!